nvtabular/dask/io.py

```diff
@@ -359,14 +374,23 @@ class CSVDatasetEngine(DatasetEngine):
     def __init__(self, *args, **kwargs):
         super().__init__(*args)
         self._meta = {}
-        self.names = kwargs.pop("names", None)
         self.csv_kwargs = kwargs
+        self.names = self.csv_kwargs.get("names", None)
```

Comment:
I don't feel too strongly about it, but there is really no reason to create a `names` attribute.  We are already storing `csv_kwargs`, which is expected to include `names`.  Note that the existing `test_dask_workflow_api_dlrm` test is already passing `names` to the dask_cudf.read_csv backend this way

Reply:
@rjzamora I felt like since both to_iter and to_dd were using it, it would be easier to set in one place. But if we want that changed, I can make that happen no problem.